src/modules/libraryModule.js

```diff
@@ -14,80 +14,127 @@ export default class Library {
+      <div class="handle removeButton" id="${id}" type="button"></div>
         `;
   }
 
   pullFromStorage = () => {
-    const parsed = JSON.parse(sessionStorage.getItem('strShelf'));
+    const parsed = JSON.parse(localStorage.getItem('strShelf'));
     let counter = 1;
     const preShelf = [];
-    for (let i = 0; i < this.shelf.length; i += 1) {
+    // Returns the length of the localstorage shelf
+    // (even when the page reloads)
+    const shelfSize = Object.values(parsed).length;
+
+    for (let i = 0; i < shelfSize; i += 1) {
+      // Returns the Nth item saved on the localstorage
+      if (this.shelf.length === 0) {
+        for (let i = 0; i < shelfSize; i += 1) {
+          this.shelf.push(Object.values(parsed)[i]);
+        }
+      }
+      // Get the completed value of parsed book
       const parsedBook = parsed[`${counter}`];
-      const tempShelf = Library.displayBook(parsedBook.title, counter);
-      preShelf.unshift(tempShelf);
+      let tempShelf = Library.displayBook(parsedBook.title, counter);
+      if (parsedBook.completed) {
+        tempShelf = Library.displayCheckedBook(parsedBook.title, counter);
+      }
+      preShelf.push(tempShelf);
       counter += 1;
     }
+    return preShelf;
+  }
 
+  paintToPage = () => {
+    const preShelf = this.pullFromStorage();
     this.frontShelf.innerHTML = '';
-
     for (let i = 0; i < preShelf.length; i += 1) {
       const createdBook = document.createElement('div');
       createdBook.classList.add('unchecked-slot', 'book');
       createdBook.innerHTML = preShelf[i];
       createdBook.id = i + 1;
       this.frontShelf.appendChild(createdBook);
     }
+    this.setRemoveListeners();
   }
 
   updateShelf = () => {
+    // console.log('bookshelf A: ' + this.bookshelf);
     this.bookshelf = new StrShelf();
+
     let counter = 0;
     for (let i = 0; i < this.shelf.length; i += 1) {
       counter += 1;
       this.shelf[i].id = counter;
+      // this.shelf[i].index = counter;
     }
     for (let i = 0; i < this.shelf.length; i += 1) {
       // GIVES THE BOOK OBJ AN ID #NUMBER
       this.bookshelf[`${this.shelf[i].id}`] = this.shelf[i];
     }
+    // console.log('bookshelf C: ' + this.bookshelf);
     Library.pushToStorage(this.bookshelf);
   }
 
   newBook = (title) => {
     this.bookshelf = new StrShelf();
     const book = new Book();
     book.title = title;
-    this.shelf.unshift(book);
+    book.completed = false;
+    this.shelf.push(book);
     this.updateShelf();
+    this.paintToPage();
+    this.setInputFieldListeners();
+    this.setCheckboxFieldListeners();
   }
 
   removeBook = (id) => {
-    this.bookshelf = new StrShelf();
+    for (let i = 0; i < this.shelf.length; i += 1) {
+      if (this.shelf[i].id === parseInt(id, 10)) {
+        this.shelf.splice(i, 1);
+        this.updateShelf();
+      }
+      this.paintToPage();
+      this.setCheckboxFieldListeners();
+    }
+  }
```

Comment:
- [ ] Hi @mrjuda , please make sure that your project is working as pointed out by the previous reviewer. Make sure that the following functionalities can be accomplished: 
- [ ] Add todos
- [ ] Delete todos
- [ ] Edith todos

I think you haven't yet resolved this issue.  Check the screenshot below:
![Screenshot from 2022-03-27 04-54-01](https://user-images.githubusercontent.com/63087491/160266208-fb05ab0f-e188-42a9-8159-ba83b2378bf7.png)

Reply:
Done it, thank you

Reply:
I already implemented the "clear all" functionality together with the "CRUD" activity (the previous PR) for the Wednesday's activity.